tests/test_api_circuit_breaker.py

```diff
@@ -0,0 +1,101 @@
+import responses
+import pytest
+import re
+from freezegun import freeze_time
+from app import app
+from apistar.test import TestClient
+from idunn.blocks.wikipedia import WikipediaBreaker
+from time import sleep
+
+from .test_full import fake_all_blocks
+
+@pytest.fixture(scope="module", autouse=True)
+def breaker_test():
+    """
+    We define here settings specific to tests
+    for the circuit_breaker in order to avoid
+    any waste of time with real timeout and
+    failmax
+    """
+    WikipediaBreaker.init_breaker()
+    wiki_breaker = WikipediaBreaker.get_breaker()
+    wiki_breaker.fail_max = 3
+    wiki_breaker.reset_timeout = 1
+    return wiki_breaker
+
+def test_circuit_breaker_500(fake_all_blocks, breaker_test):
+    """
+    Test when the external service returns a
+    HTTPError 500.
+    Test that all possible blocks are correct even
+    if the circuit is open.
+    """
+    client = TestClient(app)
+    with responses.RequestsMock() as rsps:
+        rsps.add('GET',
+             re.compile('^https://.*\.wikipedia.org/'),
+             status=500)
+        """
+        We mock 40 calls to wikipedia while the max number
+        of failure is 30, so we test that after 30 calls
+        the circuit is open (i.e there are no more than 30
+        calls sent).
+        """
+        breaker_test.close()
```

Comment:
by curiosity, how could the circuit be open there ? since we don't want side effects between test I don't feel  safe having this kind of stuff here

Reply:
You are right: in the current tests order it cannot be open.
However I thought safer to make each test independent of the order: if later we add a test with the breaker before the first one ('test_circuit_breaker_500') it could let the circuit open and make the successful 'test_circuit_breaker_500' failed (which is exactly what happens to me ^^).
So I just close it at the beginning of each test.

Reply:
hum i'm not convinced, it's easy to slip by.

maybe close it in the fixture (and make it a yield fixture)

Reply:
done